tests/ui/undropped_manually_drops.rs

```diff
@@ -0,0 +1,24 @@
+#![warn(clippy::undropped_manually_drops)]
+
+struct S;
+
+fn main() {
+    let f = std::mem::drop;
+    let g = std::mem::ManuallyDrop::drop;
+    let mut manual1 = std::mem::ManuallyDrop::new(S);
+    let mut manual2 = std::mem::ManuallyDrop::new(S);
+    let mut manual3 = std::mem::ManuallyDrop::new(S);
+    let mut manual4 = std::mem::ManuallyDrop::new(S);
+
+    // These lines will not drop `S` and should be linted
+    drop(std::mem::ManuallyDrop::new(S));
+    drop(manual1);
+    f(manual2);
```

Comment:
Just add a `FIXME` here and write this in the `Known Problems` section. I don't think binding `mem::drop` and then using it is a common enough use case for us to take care of it.

Reply:
Just added those two notes.